torchmetrics/metric.py

```diff
@@ -70,25 +71,46 @@ class Metric(Module, ABC):
         dist_sync_on_step:
             Synchronize metric state across processes at each ``forward()``
             before returning the value at the step.
+
+            .. deprecated:: v0.8
+                Argument is deprecated and will be removed in v0.9 in favour of instead
+                passing it in as keyword argument.
+
         process_group:
             Specify the process group on which synchronization is called.
             default: `None` (which selects the entire world)
+
+            .. deprecated:: v0.8
+                Argument is deprecated and will be removed in v0.9 in favour of instead
+                passing it in as keyword argument.
+
         dist_sync_fn:
             Callback that performs the allgather operation on the metric state. When `None`, DDP
             will be used to perform the allgather.
+
+            .. deprecated:: v0.8
+                Argument is deprecated and will be removed in v0.9 in favour of instead
+                passing it in as keyword argument.
+
+        metric_kwargs: additional keyword arguments
+
+            - dist_sync_on_step: If metric state should synchronize on ``forward()``
+            - process_group: The process group on which the synchonization is called
+            - dist_sync_fn: function that performs the allgather option on the metric state
     """
 
     __jit_ignored_attributes__ = ["device"]
     __jit_unused_properties__ = ["is_differentiable"]
     is_differentiable: Optional[bool] = None
     higher_is_better: Optional[bool] = None
+    dist_sync_on_step: bool = False
+    process_group: Optional[ProcessGroup] = None
+    dist_sync_fn: Optional[Callable] = None
```

Comment:
I think we cannot put them here, as these are static. Meaning that This way, I cannot have two instances of the same metric for different process goups or something like that. They need to be moved inside the `__init__`

Reply:
True. Will change.